docx/oxml/text/run.py

```diff
@@ -90,14 +90,25 @@ def text(self):
         equivalent.
         """
         text = ''
-        for child in self:
-            if child.tag == qn('w:t'):
-                t_text = child.text
-                text += t_text if t_text is not None else ''
-            elif child.tag == qn('w:tab'):
-                text += '\t'
-            elif child.tag in (qn('w:br'), qn('w:cr')):
-                text += '\n'
+        if CT_FldChar.numOfNestedFldChar > 0:
+            for child in self:
```

Comment:
For faster iteration, you can leverage `lxml` here. Example:
```python
for el in self.iterchildren(tag=qn("w:fldChar")):
    if el.get("fldCharType") == "begin":
        ...
```
This will create python objects only for `w:fldChar` elements instead of for every child.
This is not necessary here, since complex fields don't have a lot of children, but I wanted to leave a comment for future reference.

https://lxml.de/api/lxml.etree._Element-class.html#iterchildren

Reply:
Thank you for the suggestion.